app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDao.java

```diff
@@ -89,6 +89,7 @@ public UserConsent getActiveUserConsent(String healthCode, SubpopulationGuid sub
 
         DynamoDBQueryExpression<DynamoUserConsent3> query = new DynamoDBQueryExpression<DynamoUserConsent3>()
             .withScanIndexForward(false)
+            .withLimit(1)
             .withQueryFilterEntry("withdrewOn", new Condition().withComparisonOperator(ComparisonOperator.NULL))
             .withHashKeyValues(hashKey);
```

Comment:
We check twice in the consent path to make sure they aren't creating a duplicate, including right before creating the user consent record. So it should not be possible but you could contrive it to cause it to happen. However, externally it will appear you are withdrawn because you can only have one signed and active signature. This is a bucket of worms, I can write a test to see if, for example, you are considered consented at this point or not.

If we just moved the consentCreatedOn value into the signature, we could get rid of the UserConsent table and then there would be no danger of inconsistency. They would either have an active signature in their history, or not.